mitodl/micromasters
# Add mail_id and template variables to Mailgun functions

#### What are the relevant tickets?
Part of #2241 
Fixes #3035 

#### What's this PR do?
 - Adds a UUID field to Profile
 - Uses that UUID and `first_name` in the template variables for automatic emails and bulk mail sent out via search.

#### How should this be manually tested?
Send an email with the email campaign option set and `MAILGUN_RECIPIENT_OVERRIDE` set to your email. Verify that you see a UUID and a first name in the body of the email.

Next, change or create a new Profile which matches the criteria for the automatic email. Verify that a new email was sent just for that user, also containing their template variables.
